### PR TITLE
Reorganizing of test hosts across library

### PR DESCRIPTION
**Pull request by timotheeguerin**
Was getting started to look at adding more tests for decorators as part of https://github.com/Azure/cadl-azure/issues/827 and our current usage of test host in libraries is a bit of a mess. Trying to simplify this.

Redesign a bit the test host to be able to better reuse other libraries test hosts without having to manually redefine all the files to import.

Main changes:
- `/testing` scope to packages so you can import this with the test utils
- each package can provide a `TestLibrary` that provide the files needeed to use that library. Another library can then depend on it without having to manually reference every single paths.
- Provide helper expectations for diagnostics
```ts
expectDiagnostics(diagnostics, [{ code: "suppress-error" }, { code: "no-id-property" }]);
```
image

### CODE REVIEW

File: packages/compiler/testing/test-host.ts
Comment (daviwil) [inline packages/compiler/testing/test-host.ts:178]: `resolveVfsPath` or `resolveVFSPath`?

**Review by daviwil — APPROVED** [review]
This is so much better, thanks for doing it!

File: packages/compiler/testing/test-host.ts
Comment (nguerrera) [reply packages/compiler/testing/test-host.ts:178]: I renamed this to resolveVirtualPath in #207 :) I was holding that back to avoid merge conflicts, but regretting my choice now, lol.

Comment (timotheeguerin) [reply packages/compiler/testing/test-host.ts:178]: I'll wait until the projection and your Pr are merged so I resolve that conflict

Comment (nguerrera) [reply packages/compiler/testing/test-host.ts:178]: Thanks, and I'm waiting on your emit PR too.

I think fair FIFO order is

1. projections 
2. emit
3. testable LS
4. this PR

cc @bterlson

Comment (timotheeguerin) [reply packages/compiler/testing/test-host.ts:178]: sounds good

**Comment by timotheeguerin** [comment]
@nguerrera merged all the others PRs, cleanup and added a helper to test diagnostics  if you want to take another look at this PR. It should be good now

**Review by nguerrera — APPROVED** [review]

**Pull request merged**
